src/main/java/org/sunbird/migrate/service/impl/UserMigrationServiceImpl.java

```diff
@@ -0,0 +1,247 @@
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+
+@Service
+public class UserMigrationServiceImpl implements UserMigrationService {
+
+    private Logger log = LoggerFactory.getLogger(getClass().getName());
+
+    @Autowired
+    OutboundRequestHandlerServiceImpl outboundRequestHandlerService;
+
+    @Autowired
+    CbExtServerProperties serverConfig;
+
+    @Autowired
+    PropertiesConfig propertiesConfig;
+
+    @Autowired
+    ProfileService profileService;
+
+    @Override
+    public SBApiResponse migrateUsers() {
+        SBApiResponse response = new SBApiResponse(Constants.API_USER_MIGRATION);
+        StringBuilder url = new StringBuilder(propertiesConfig.getLmsServiceHost()).append(propertiesConfig.getLmsUserSearchEndPoint());
+        log.info("printing user search Url: {}", url);
+
+      try {
+          Map<String, Object> request = userSearchRequestBody();
+          Map<String, Object> updateResponse = outboundRequestHandlerService.fetchResultUsingPatch(url.toString(), request, null);
+
+          if (Constants.OK.equalsIgnoreCase((String) updateResponse.get(Constants.RESPONSE_CODE))) {
+              Map<String, Object> result = (Map<String, Object>) updateResponse.get(Constants.RESULT);
+              Map<String, Object> responseData = (Map<String, Object>) result.get(Constants.RESPONSE);
+              List<Map<String, Object>> users = (List<Map<String, Object>>) responseData.get(Constants.CONTENT);
+              String targetOrgName = Constants.IGOT;
+              boolean orgFound = false;
+              boolean allOperationsSuccessful = true;
+
+              for (Map<String, Object> user : users) {
+                  List<Map<String, Object>> organisations = (List<Map<String, Object>>) user.get(Constants.ORGANISATIONS);
+                  String userId = (String) user.get(Constants.USER_ID);
+                  if (organisations != null) {
+                      orgFound = organisations.stream()
+                              .anyMatch(organisation -> targetOrgName.equalsIgnoreCase((String) organisation.get(Constants.ORG_NAME)));
+                      if (!orgFound) {
+                          log.info("Organization '{}' not found for user ID '{}'. Initiating migration API call.", targetOrgName, userId);
+                          String errMsg = executeMigrateUser(getUserMigrateRequest(userId, targetOrgName, false), null);
+                          if (StringUtils.isNotEmpty(errMsg)) {
+                              log.info("Migration failed for user ID '{}'. Error: {}", userId, errMsg);
+                              allOperationsSuccessful = false;
+                          } else {
+                              log.info("Successfully migrated user ID '{}'.", userId);
+                              Map<String, Object> profileDetails = (Map<String, Object>) user.get(Constants.PROFILE_DETAILS);
```

Comment:
Do not use this profileDetails
extPatch method needs it's own newly created profileDetails -- which should have attributes which are needed update... not all the fields...